tests/integration/callbacks/test_pattern_matching.py

```diff
@@ -0,0 +1,374 @@
+  ),
+  mark_done <- function(done){
+    if (length(done[[1]] > 0)) return(style_done) else return(style_todo)
+  }
+)
+
+
+app$callback(
+  output(id = "totals", property = "children"),
+  params = list(
+    input(list("index" = ALL, "type" = "done"), property = "value"),
+    state(list("index" = ALL, "type" = "check"), property = "children")
+  ),
+  show_totals <- function(done, total) {
+    count_all = length(total)
+    count_done = length(done)
+
+    result = sprintf("%s of %s items completed", count_done, count_all)
+
+    if (count_all > 0) {
+      result = paste(result, sprintf(" - %s%%", as.integer(100 * count_done/count_all)))
+    }
+
+    if (is.null(total[[1]])) {
+      return("Add an item to the list to get started.")
+    } else {
+      return(result)
+    }
+  }
+)
+
+
+app$run_server()
+"""
+
+
+def test_rpmc001_pattern_matching_all(dashr):
+    dashr.start_server(all_app)
+    dashr.find_element("#add-filter").click()
+    dashr.select_dcc_dropdown('#\\{\\"bar\\"\\:1\\,\\"foo\\"\\:\\"filter-dropdown\\"\\}', "NYC")
+    dashr.wait_for_text_to_equal(
+        "#dropdown-container-output",
+        "Dropdown 1 = NYC"
+    )
+    dashr.find_element("#add-filter").click()
+    dashr.select_dcc_dropdown('#\\{\\"bar\\"\\:2\\,\\"foo\\"\\:\\"filter-dropdown\\"\\}', "MTL")
+    dashr.wait_for_text_to_equal(
+        "#dropdown-container-output",
+        "Dropdown 1 = NYC"
+        "\n"
+        "Dropdown 2 = MTL"
+    )
+
+
+def test_rpmc002_pattern_matching_allsmaller(dashr):
+    dashr.start_server(allsmaller_app)
+    dashr.find_element("#add-filter-ex3").click()
+    dashr.select_dcc_dropdown('#\\{\\"index\\"\\:1\\,\\"type\\"\\:\\"filter-dropdown-ex3\\"\\}', "Argentina")
+    dashr.find_element("#add-filter-ex3").click()
+    dashr.select_dcc_dropdown('#\\{\\"index\\"\\:2\\,\\"type\\"\\:\\"filter-dropdown-ex3\\"\\}', "Angola")
+    dashr.wait_for_text_to_equal(
+        "#test",
+        "59.03 is the life expectancy of Angola and Argentina."
+    )
+
+
+def test_rpmc003_pattern_matching_match(dashr):
+    dashr.start_server(match_app)
+    dashr.find_element("#dynamic-add-filter").click()
+    dashr.select_dcc_dropdown('#\\{\\"index\\"\\:1\\,\\"type\\"\\:\\"dynamic-dropdown\\"\\}', "NYC")
+    dashr.wait_for_text_to_equal(
+        '#\\{\\"index\\"\\:1\\,\\"type\\"\\:\\"dynamic-output\\"\\}',
+        "Dropdown 1 = NYC"
+    )
+    dashr.find_element("#dynamic-add-filter").click()
+    dashr.select_dcc_dropdown('#\\{\\"index\\"\\:2\\,\\"type\\"\\:\\"dynamic-dropdown\\"\\}', "MTL")
+    dashr.wait_for_text_to_equal(
+        '#\\{\\"index\\"\\:2\\,\\"type\\"\\:\\"dynamic-output\\"\\}',
+        "Dropdown 2 = MTL"
+    )
+
+
+def test_rpmc004_pattern_matching_todo(dashr):
+    dashr.start_server(todo_app)
+    dashr.find_element("#new-item").send_keys("Item 1")
+    dashr.find_element("#add").click()
+    dashr.find_element("#new-item").send_keys("Item 2")
+    dashr.find_element("#add").click()
```

Comment:
```suggestion
    dashr.find_element("#add").click()
    dashr.wait_for_element_by_css_selector('#\\{\\"index\\"\\:1\\,\\"type\\"\\:\\"done\\"\\}')
```
Minor fix for test flakiness.

Reply:
@HammadTheOne :bow: